clang/test/lab1/simonyan_suren/test.cpp

```diff
@@ -0,0 +1,54 @@
+// RUN: %clang_cc1 -ast-dump -ast-dump-filter testFun -load %llvmshlibdir/SimonyanAlwaysInlinePlugin%pluginext -add-plugin always-inlines-plugin %s 2>&1 | FileCheck %s
+
+// CHECK: FunctionDecl {{0[xX][0-9a-fA-F]+ <.+test\.cpp:([0-9]+:[0-9]|[0-9]+), (line|col):([0-9]+:[0-9]|[0-9]+)> (line|col):([0-9]+:[0-9]|[0-9]+) testFun1 'int \(int, int\)'}}
+int testFun1(int A, int B) {
+  if (A < B) {
+    A = B;
+  }
+  return 0;
+}
+// CHECK-NOT: `-AlwaysInlineAttr {{0[xX][0-9a-fA-F]+ <(line|col):([0-9]+:[0-9]|[0-9]+)> Implicit always_inline}}
+
+// CHECK: FunctionDecl {{0[xX][0-9a-fA-F]+ <.+test\.cpp:([0-9]+:[0-9]|[0-9]+), (line|col):([0-9]+:[0-9]|[0-9]+)> (line|col):([0-9]+:[0-9]|[0-9]+) testFun2 'void \(int, int\)'}}
+void testFun2(int A, int B) {
+  {
+    while(A < B) {
+      A = B;
+    }
+  }
+}
+// CHECK-NOT: `-AlwaysInlineAttr {{0[xX][0-9a-fA-F]+ <(line|col):([0-9]+:[0-9]|[0-9]+)> Implicit always_inline}}
+
+// CHECK: FunctionDecl {{0[xX][0-9a-fA-F]+ <.+test\.cpp:([0-9]+:[0-9]|[0-9]+), (line|col):([0-9]+:[0-9]|[0-9]+)> (line|col):([0-9]+:[0-9]|[0-9]+) testFun4 'void \(char, char\)'}}
+void testFun4(char A, char B) {
+  {
+    {
+      while (true) {
+      }
+    }
+  }
+}
+// CHECK-NOT: `-AlwaysInlineAttr {{0[xX][0-9a-fA-F]+ <(line|col):([0-9]+:[0-9]|[0-9]+)> Implicit always_inline}}
+
+// CHECK: FunctionDecl {{0[xX][0-9a-fA-F]+ <.+test\.cpp:([0-9]+:[0-9]|[0-9]+), (line|col):([0-9]+:[0-9]|[0-9]+)> (line|col):([0-9]+:[0-9]|[0-9]+) testFun5 'int \(int\)'}}
+int testFun5(int A) {
+  for (int i = 0; i < 0; i++){
+  }
+
+  return A;
+}
+// CHECK-NOT: `-AlwaysInlineAttr {{0[xX][0-9a-fA-F]+ <(line|col):([0-9]+:[0-9]|[0-9]+)> Implicit always_inline}}
+
+// CHECK: FunctionDecl {{0[xX][0-9a-fA-F]+ <.+test\.cpp:([0-9]+:[0-9]|[0-9]+), (line|col):([0-9]+:[0-9]|[0-9]+)> (line|col):([0-9]+:[0-9]|[0-9]+) testFun6 'int \(int\)'}}
+int testFun6(int A) {
```

Comment:
Could you please also add positive cases with functions without conditional instructions?